src/IR.cpp

```diff
@@ -652,6 +652,12 @@ const char *const intrinsic_op_names[] = {
     "widening_shift_left",
     "widening_shift_right",
     "widening_sub",
+    "make_struct_type",
```

Comment:
Should this be renamed? It's odd for an expression to return a type. Maybe make_null_typed_struct?

Reply:
I renamed it ```declare_struct_type``` which is perhaps more accurate. It returns a value of the type, which can be used to apply the type to a different value later. Whether this is odd or not probably depends on one's viewpoint, but it is a way for the backend type to go from one place to another without Halide IR knowing about it and is also the sort of thing that shows up in template metaprogramming a fair bit.